android/app/build.gradle

```diff
@@ -127,14 +127,14 @@ def jscFlavor = 'org.webkit:android-jsc:+'
  */
 def enableHermes = project.ext.react.get("enableHermes", false);
 
+/**
+ * Architectures to build native code for in debug.
+ */
+def nativeArchitectures = project.getProperties().get("reactNativeDebugArchitectures")
+
 android {
     compileSdkVersion rootProject.ext.compileSdkVersion
 
-    compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
-    }
-
```

Comment:
Unless there's a good reason for this I wouldn't want to make this change. This ensures we all compile the app [with a consistent Java version](https://developer.android.com/reference/tools/gradle-api/4.1/com/android/build/api/dsl/CompileOptions#sourceCompatibility:org.gradle.api.JavaVersion).

Reply:
I assume there has to be a good reason for it to be listed in the upgrade helper, but it would be good to confirm this first.

Reply:
Okay, I had to dig a bit but here is [the commit](https://github.com/facebook/react-native/commit/547b4c92e4743f5b5816297f48a608ace9de6bb5) that removes this code from the template. It was removed because gradle 4.2 defaults to Java 8, so this is not necessary anymore.

Reply:
Cool, thanks for confirming.